### PR TITLE
Fix #1971

### LINKED ISSUE #1971: Metric relabeling capability in the helm chart
**Describe the solution you'd like**

Ability to set further attributes of the servicemonitor resource

**What is the added value?**

While running in a multi-tenant setup, we can have multiple alertmanagerconfig resources (typically one per namespace) of the kube-prometheus-stack which has an implicit matcher on the namespace label. Given the namespace label in this case is always the namespace of the service monitor resource and the value of the namespace of external secret resource is under the exported_namespace label. 

**Give us examples of the outcome**

Desired endpoints block in servicemonitor spec:
```yaml
  endpoints:
  - interval: 30s
    port: metrics
    scrapeTimeout: 25s
    honorLabels: true
    metricRelabelings:
    - action: replace
      regex: (.*)
      replacement: $1
      sourceLabels:
      - exported_namespace
      targetLabel: namespace
```
**Observations (Constraints, Context, etc):**

Current scenario
Screenshot 2023-02-01 at 10 24 00

Desired scenario
Screenshot 2023-02-01 at 10 34 05

### PR DESCRIPTION
**Pull request by mikhailadvani**
This PR helps fix #1971 by allowing the user to override the relabeling configurations. Default values are blank to ensure backward compatibility

### CODE REVIEW

File: deploy/charts/external-secrets/values.yaml [hunk 1]
```diff
@@ -126,6 +126,27 @@ serviceMonitor:
   # -- Timeout if metrics can't be retrieved in given time interval
   scrapeTimeout: 25s
 
+  # -- Let prometheus add an exported_ prefix to conflicting labels
+  honorLabels: true
```
Comment (moolen) [inline deploy/charts/external-secrets/values.yaml:130]: I think this has been false prior this PR (this seems to be the default). Can we change this to `false`?

File: deploy/charts/external-secrets/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -36,4 +36,13 @@ spec:
   - port: metrics
     interval: {{ .Values.serviceMonitor.interval }}
     scrapeTimeout: {{ .Values.serviceMonitor.scrapeTimeout }}
+    honorLabels: {{ .Values.serviceMonitor.honorLabels }}
+    {{- with .Values.serviceMonitor.metricRelabelings }}
+    metricRelabelings:
+      {{- tpl (toYaml . | nindent 6) . }}
```
Comment (moolen) [inline deploy/charts/external-secrets/templates/servicemonitor.yaml:42]: Why do you need to `tpl` here? I'm curious about the use-case of this.

**Review by moolen — COMMENTED** [review]
Looks good, especially having comments that points to docs is really helpful!
On comment regarding defaults of `honorLabels`.

File: deploy/charts/external-secrets/values.yaml (hunk 1 above)
Comment (mikhailadvani) [reply deploy/charts/external-secrets/values.yaml]: fixed

File: deploy/charts/external-secrets/templates/servicemonitor.yaml (hunk 2 above)
Comment (mikhailadvani) [reply deploy/charts/external-secrets/templates/servicemonitor.yaml]: copy paste errors. fixed

**Review by moolen — APPROVED** [review]
LGTM, thank your for this :medal_sports:

**Pull request merged**